src/angular/popup-menu/popup-menu-list.component.ts

```diff
@@ -45,6 +45,7 @@ export class PopupMenuListComponent implements AfterContentInit {
         }
     }
     @Input() public className: any;
+    @Input() public relative: boolean;
```

Comment:
Since `relative` is only used for CSS, you could omit this Input, and instead use `className` for it.
By default, popup-menu will be styled as static, and only when having `.relative` in className, it will override the CSS to turn it into relative.

Note: `left` and `top` CSS styles are not effective when `position: static`.
Or maybe (even better), you can put the `.relative` class automatically when `position` is transferred and is not null/undefined.

Same issue goes for React implementation.

Reply:
I spoke with Israel about this as well. We agreed that using a prop instead of className is the correct approach. "Relative" is the type of the menu. You should never use className to determine type or achieve functionality in a component. The component should receive props and use them to take care of the classes used on its own. 

You can read about this here: https://brigade.engineering/don-t-pass-css-classes-between-components-e9f7ab192785

Reply:
I'm not fully agree with that article. You can always wrap a component and override its style even without using className, so not using className doesn't really avoid complexity. (In Angular component styles can be isolated, and thus not be able to override them).
className is convenient to combine some styles together. Otherwise you can end up with many properties, which is quite equivalent in complexity but many props may hurt performance sometimes, in managing them all.
If a property is not used for "special" functionality inside the component, but only for CSS - it can end up with a className passed to the component, and exposing the available classNames is like exposing the available properties.
I believe in using className - let the developer decide what he wants - stick to the defaults styles or custom it for some special cases.